### ISSUE TITLE
Rubberduck crashing during first parse.

### ISSUE THREAD

**Issue by SteveLaycock** [rubberduck-vba/Rubberduck#5217]
**Rubberduck version information**

   Version 2.4.1.5071
OS: Microsoft Windows NT 10.0.18362.0, x64
Host Product: Microsoft Office x64
Host Version: 16.0.12026.20320
Host Executable: WINWORD.EXE

**Description**
Rubberduck crashes during the 'Loading' phase of the first parse.  There is no log produced.  Version 69 was fine, the first and subsequent parses complete without issues. Crashing started in 71 and is repeatable in 72.

**Comment by bclothier** [comment]
The 69 corresponds to PR #5211. Presumably the fix is not stable and should be reverted

**Comment by bclothier** [comment]
I just tried to replicate this and failed. I even opened the battelship XLSM and also tried Word but I did not crash. 

Can you confirm it's not specific to a certain Word document?

**Comment by bclothier** [comment]
To help us test whether the PR #5211 is truly at fault, you can download this:

https://ci.appveyor.com/api/buildjobs/rmsfcmucy3l36k7x/artifacts/Rubberduck.Deployment%2FInnoSetup%2FInstallers%2FRubberduck.Setup.2.4.1.5075-pre.exe

**Comment by SteveLaycock** [comment]
Sorry for not responding sooner but I was out accompanying my SO on her new electric bike,.

5075 and 5077 still crash.

I reinstalled 5069 but now that is also crashing and I have to go back to 5061 to get stable behaviour.

5067 crashes but has other odd behaviours in that it doen't crash every time, but very occasionally produces an 'Unexpected error' parse message.

I have also done multiple reboots and rebuilds for Microsoft office.

**Comment by SteveLaycock** [comment]
Seems that I recently reset Rubberduck settings as no logging was enabled <blush>.  Logg from 5077 is attached.
[RubberduckLog.txt](https://github.com/rubberduck-vba/Rubberduck/files/3721950/RubberduckLog.txt)

**Comment by MDoerner** [comment]
The crash seems to happen in the code loading references. At least that is where the log stops.
Could you tell us which libraries you reference when you experience the crash?

**Comment by SteveLaycock** [comment]
![references](https://user-images.githubusercontent.com/24568897/66719280-a8651000-ede5-11e9-80d3-f1561844a61c.jpg)

**Comment by bclothier** [comment]
@SteveLaycock  can you confirm this no longer crash?

https://ci.appveyor.com/api/buildjobs/lsp21631c2tg96a5/artifacts/Rubberduck.Deployment%2FInnoSetup%2FInstallers%2FRubberduck.Setup.2.4.1.5082-pre.exe

Thanks!

**Comment by SteveLaycock** [comment]
I'm afraid I'm still  seeing crashes with 5081.  5061 is running fine.  Log for 5081 attached.  If you need to see my code I'm happy to provide.
[RubberduckLog.txt](https://github.com/rubberduck-vba/Rubberduck/files/3726355/RubberduckLog.txt)

**Comment by bclothier** [comment]
Wait a minute, @SteveLaycock  that's 5081. The link I indicated above is 5082. Can you try that, please?

**Comment by SteveLaycock** [comment]
@bclothier Oh I misread your post.  I'm happy to confirm that 5082 completes the first and subsequent parses without issue

**Comment by bclothier** [comment]
Thanks for confirming! I'll merge the hotfix then.

**Comment by bclothier** [comment]
Hi, I'm trying to reproduce the crash so I can improve the handling around this -- I could add all references except for two and not crash. Can you tell me where I'd get the `TypeLib Information` and `VBAExtensions: Better Objects for VBA`?

Thanks!

**Comment by SteveLaycock** [comment]
TypeLib Information is C:\Windows\SysWow64\TLBInf32.dll

![TypeLinInfo](https://user-images.githubusercontent.com/24568897/67140544-82ab9100-f253-11e9-98c2-e3a46967d4ca.jpg)

TBH I think this reference is there only because of a past exploration and could be removed.

VBAExtensions is my 'delusion of adequacy'.  It is a dictionary class  (Kvp  (SmartKvp)). I wrote in c#.  A zip of the Visual Studio (2019) project is attached.  Please note that this extension was my first foray into c# and that I'm not a professional programmer.

[RegCmcLib.zip](https://github.com/rubberduck-vba/Rubberduck/files/3746428/RegCmcLib.zip)

**Comment by bclothier** [comment]
Dammit, no reproduction here. I was so sure that the `TLBINF32.DLL` would be the culprit. The thing is that this particular library was never upgraded to 64-bit and you're running a 64-bit Word, but referencing a 32-bit Type Library in `C:\Windows\SysWOW64\` (read: a folder of 32-bit DLLs on a 64-bit OS). 

I used the version 71 in order to reproduce the crash, and even made a blank Word document with all the references in same order (I did not know how to remove the `Normal` reference):
![image](https://user-images.githubusercontent.com/2367644/67146467-f4361000-f250-11e9-810e-9144cf70e75b.png)

This compiles and can parse without any error. I tried adding procedures that has variables from both `TypeLib Information` and `VBAExtensions` and that still didn't crash.

Do you think you could try and reinstall the version 71, remove the `TypeLib Information` reference and see if that avoids crashing?

**Comment by SteveLaycock** [comment]
Will do. But it will be at least 3 hours before I can get back to my pc.

**Comment by SteveLaycock** [comment]
OK I tested with 5071 and as previously Word crashed.

To help things along I've attached a copy of the word template.  The template also contains extensive customisation of the Ribbon via CustomUI.  As this template is a rewrite of something I did previously, the ribbon callbacks are not yet implemented you you get a number of errors during startup.  This is a personal project so there is nothing confidential to worry about.

[RegulatoryCMC - Copy.zip](https://github.com/rubberduck-vba/Rubberduck/files/3746964/RegulatoryCMC.-.Copy.zip)
